### PR TITLE
PS-3958: handle_fatal_signal (sig=11) in subselect_hash_sj_engine::cl…

### PR DESCRIPTION
**Pull request by gl-sergei**
…eanup

subselect_hash_sj_engine::cleanup crashed when temporary table was not
created.

Fix is to avoid crash.

Changes in MTR required to allow it to start server with "Encryption
can't find master key" error.

### CODE REVIEW

**Comment by gl-sergei** [comment]
https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1789/

File: sql/item_subselect.cc [hunk 1]
```diff
@@ -4013,15 +4013,20 @@ void subselect_hash_sj_engine::cleanup()
...
+    result->cleanup(); /* Resets the temp table as well. */
   THD * const thd= item->unit->thd;
   DEBUG_SYNC(thd, "before_index_end_in_subselect");
-  TABLE *const table= tab->table();
-  if (table->file->inited)
-    table->file->ha_index_end();  // Close the scan over the index
-  free_tmp_table(thd, table);
-  // Note that tab->qep_cleanup() is not called
-  tab= NULL;
+  fprintf(stderr, "tab = %p\n", tab);
```
Comment (laurynas-biveinis) [inline sql/item_subselect.cc]: Remove

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
LGTM with fprintf removed, no need for another review.

File: mysql-test/t/ps-3958.test [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#
```
Comment (laurynas-biveinis) [inline mysql-test/t/ps-3958.test]: On a second thought, should this be added to the main --innodb-temp-tablespace-encrypt testcase?

**Comment by gl-sergei** [comment]
https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1798/

Test case merged into temp_table_encrypt.inc

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**